src/components/network/network.js

```diff
@@ -607,15 +607,17 @@ export default class Network {
                 break;
             case 'VOLTAGE_LEVEL':
                 const substationId =
-                    this.voltageLevelsById.get(equipmentId).substationId;
-                let voltageLevelsOfSubstation =
-                    this.substationsById.get(substationId).voltageLevels;
-                voltageLevelsOfSubstation = voltageLevelsOfSubstation.filter(
-                    (l) => l.id !== equipmentId
-                );
-
-                this.substationsById.get(substationId).voltageLevels =
-                    voltageLevelsOfSubstation;
+                    this.voltageLevelsById.get(equipmentId)?.substationId;
+                const substation = this.substationsById.get(substationId);
+                if (substation) {
```

Comment:
It could be:
if (substation) {
                    substation.voltageLevels = substation.voltageLevels.filter(
                        (vl) => vl.id !== equipmentId
                    );
                }